Integration test part demo and integration test script for traffic label transparent transmission plugin.

daizhenyu opened the pull request:
【Fix issue】#1333

[Modification] Added some demos and integration test scripts for the integration test of the traffic label transparent transmission plug-in, including:
1.rocketmq-producer-demo
2.kafka-consumer-demo
3.kafka-producer-demo
4.tag-transmission-integration-test
5. Integration test script

[Use case description] Traffic label transparent transmission plug-in integration test

[Self-test situation] 1. Local static check passed

[Scope of influence] None

hanbingleixue commented on sermant-integration-tests/tag-transmission-test/tag-transmission-integration-test/src/test/java/com/huaweicloud/demo/tagtransmission/integration/utils/RequestUtils.java:
```diff
@@ -0,0 +1,64 @@
+
+            // 关闭响应和 HttpClient
+            response.close();
+            httpClient.close();
```
建议在finally里面关闭资源或者使用try-with-resource

daizhenyu replied:
已修改

hanbingleixue commented on sermant-integration-tests/tag-transmission-test/tag-transmission-integration-test/src/test/java/com/huaweicloud/demo/tagtransmission/integration/TagTransmissionTest.java:
```diff
@@ -0,0 +1,262 @@
+        if (jsonOptional.isPresent()) {
+            tagMap = JSON.parseObject(jsonOptional.get(), Map.class);
+        }
+        return tagMap;
```
建议jsonOptional的值为空时再 new HashMap<>()，如下：
        if (jsonOptional.isPresent()) {
            return JSON.parseObject(jsonOptional.get(), Map.class);
        }
        return  new HashMap<>();

daizhenyu replied:
已修改

hanbingleixue commented on sermant-integration-tests/tag-transmission-test/tag-transmission-integration-test/src/test/java/com/huaweicloud/demo/tagtransmission/integration/TagTransmissionTest.java:
```diff
@@ -0,0 +1,262 @@
+                new ExponentialBackoffRetry(1000, 3));
+        curator.start();
+        String nodePath = "/sermant/tag-transmission-plugin/tag-config";
+        String lineSeparator = System.getProperty("line.separator");
```
System.getProperty("line.separator")是换行符吗？考虑是否可以使用System.lineSeparator();

daizhenyu replied:
是换行符

lilai23 commented on .github/actions/common/tag-transmission/action.yml:
```diff
@@ -0,0 +1,59 @@
+      run: |
+        tar -zxf apache-zookeeper-3.6.3-bin.tar.gz
+        bash apache-zookeeper-3.6.3-bin/bin/zkServer.sh start apache-zookeeper-3.6.3-bin/conf/zoo_sample.cfg
+    - name: download local cse
+      uses: actions/cache@v3
+      with:
+        path: Local-CSE-2.1.3-linux-amd64.zip
+        key: ${{ runner.os }}-local-cse
+        restore-keys: |
+          ${{ runner.os }}-local-cse
+    - name: start cse
+      shell: bash
+      run: |
+        export ROOT_PATH=$(pwd)
+        bash ./sermant-integration-tests/scripts/startCse.sh
```
这里应该不需要用cse吧？

daizhenyu replied:
只有servicecomb需要

daizhenyu replied:
已修改

lilai23 commented on .github/workflows/tagtransmission_integration_test.yml:
```diff
@@ -0,0 +1,445 @@
+      - name: package agent
+        run: |
+          sed -i '/sermant-backend/d' pom.xml
+          sed -i '/sermant-backend-lite/d' pom.xml
```
项目中没有sermant-backend-lite了

daizhenyu replied:
已修改

lilai23 commented on .github/actions/common/tag-transmission/action.yml:
```diff
@@ -0,0 +1,59 @@
+        java-version: ${{ env.javaVersion }}
+        distribution: 'adopt'
+        cache: maven
+    - name: download zookeeper
```
这里和下面的叫从缓存中获取比较合适，之前得描述不太准确。下载和缓存是在workflow中做的

daizhenyu replied:
已修改

lilai23 commented on .github/actions/scenarios/tag-transmission/alibaba-dubbo/action.yml:
```diff
@@ -0,0 +1,65 @@
+      shell: bash
+      run: |
+        mvn test -Dtag.transmission.integration.test.type=ALIBABA_DUBBO --file \
+        sermant-integration-tests/tag-transmission-test/tag-transmission-integration-test/pom.xml
```
pom.xml路径不对

lilai23 commented on .github/actions/scenarios/tag-transmission/apache-dubbo/action.yml:
```diff
@@ -0,0 +1,57 @@
+      shell: bash
+      run: |
+        mvn test -Dtag.transmission.integration.test.type=APACHE_DUBBO --file \
+        sermant-integration-tests/tag-transmission-test/tag-transmission-integration-test/pom.xml
```
路径也不对，下面都是，流水线测试用例似乎都没跑

lilai23 commented on .github/actions/scenarios/tag-transmission/crossthread/action.yml:
```diff
@@ -0,0 +1,50 @@
+      run: |
+        mvn package -DskipTests -Pcrossthread-test --file \
+        sermant-integration-tests/tag-transmission-test/pom.xml
+    - name: start common httpserver
+      shell: bash
+      run: |
+        nohup java -jar sermant-integration-tests/tag-transmission-test/httpserver-common-demo/target/httpserver-common-demo.jar > ${{ env.logDir }}/httpserver-common-demo.log 2>&1 &
```
公共步骤放在common里更合适

daizhenyu replied:
已修改

lilai23 commented on sermant-integration-tests/tag-transmission-test/tag-transmission-integration-test/src/test/java/com/huaweicloud/demo/tagtransmission/integration/utils/RequestUtils.java:
```diff
@@ -0,0 +1,64 @@
+            // 关闭响应和 HttpClient
+            response.close();
+            httpClient.close();
+        } catch (Exception e) {
```
异常处理

daizhenyu replied:
已修改

lilai23 commented on sermant-integration-tests/tag-transmission-test/tag-transmission-integration-test/src/test/java/com/huaweicloud/demo/tagtransmission/integration/utils/RequestUtils.java:
```diff
@@ -0,0 +1,69 @@
+                if (response.getStatusLine().getStatusCode() == 200) {
+                    return Optional.ofNullable(EntityUtils.toString(response.getEntity()));
+                } else {
+                    LOGGER.info("Request error, the message is: {}", EntityUtils.toString(response.getEntity()));
```
日志级别severe

daizhenyu replied:
已修改

lilai23 commented on sermant-integration-tests/tag-transmission-test/tag-transmission-integration-test/src/test/java/com/huaweicloud/demo/tagtransmission/integration/utils/RequestUtils.java:
```diff
@@ -0,0 +1,69 @@
+                }
+            }
+        } catch (IOException e) {
+            LOGGER.info("Request exception, the message is: {}", e.getMessage());
```
日志级别severe

daizhenyu replied:
已修改

hanbingleixue commented on sermant-integration-tests/tag-transmission-test/tag-transmission-integration-test/src/test/java/com/huaweicloud/demo/tagtransmission/integration/TagTransmissionTest.java:
```diff
@@ -0,0 +1,261 @@
+                "dynamic config transmit traffic tag failed");
+    }
+
+    private void checkTagTransmission(String url, Map<String, String> tagMap, String message) {
```
参数tagMap没有使用

daizhenyu replied:
已修改

hanbingleixue commented on sermant-integration-tests/tag-transmission-test/tag-transmission-integration-test/src/test/java/com/huaweicloud/demo/tagtransmission/integration/TagTransmissionTest.java:
```diff
@@ -0,0 +1,261 @@
+    private void checkTagTransmission(String url, Map<String, String> tagMap, String message) {
+        Map<String, String> returnTagMap = convertJson2Map(RequestUtils.get(url,
+                EXACT_TAG_MAP));
+        Assertions.assertEquals("001", returnTagMap.get("id"), message + " transmit traffic tag failed");
```
"001不应该写死，应该从tagMap中获取

daizhenyu replied:
已修改

lilai23 commented on sermant-integration-tests/tag-transmission-test/tag-transmission-integration-test/src/test/java/com/huaweicloud/demo/tagtransmission/integration/TagTransmissionTest.java:
```diff
@@ -0,0 +1,261 @@
+    @Test
+    @EnabledIfSystemProperty(named = "tag.transmission.integration.test.type", matches = "ROCKETMQ")
+    public void testRocketmq() throws InterruptedException {
+        // 生产消息
```
生产消息之前做一下校验

daizhenyu replied:
已修改

lilai23 commented on sermant-integration-tests/tag-transmission-test/tag-transmission-integration-test/src/test/java/com/huaweicloud/demo/tagtransmission/integration/TagTransmissionTest.java:
```diff
@@ -0,0 +1,261 @@
+    @EnabledIfSystemProperty(named = "tag.transmission.integration.test.type", matches = "KAFKA")
+    public void testKafka() throws InterruptedException {
+        // 生产消息
+        RequestUtils.get("http://127.0.0.1:9054/kafkaProducer/testKafkaProducer", EXACT_TAG_MAP);
```
生产消息之前做一下校验

daizhenyu replied:
已修改

luanwenfei-venus commented on .github/workflows/tagtransmission_integration_test.yml:
```diff
@@ -0,0 +1,444 @@
+      - '*'
+    paths:
+      - 'sermant-agentcore/**'
+      - 'sermant-integration-tests/**'
```
可以细化到tag-transmission

luanwenfei-venus commented on .github/workflows/tagtransmission_integration_test.yml:
```diff
@@ -0,0 +1,444 @@
+      - 'sermant-integration-tests/**'
+      - 'sermant-plugins/sermant-tag-transmission/**'
+      - '.github/workflows/tagtransmission_integration_test.yml'
+      - '.github/actions/**'
```
细化一下

lilai23 commented on .github/workflows/tagtransmission_integration_test.yml:
```diff
@@ -0,0 +1,444 @@
+name: Tagtransmission integration test
+env:
+  projectSpringBootVersion: 2.7.15
```
无用环境变量 删除

daizhenyu replied:
已修改

hanbingleixue commented on .github/actions/common/tag-transmission/action.yml:
```diff
@@ -0,0 +1,47 @@
+      run: |
+        nohup java -jar sermant-integration-tests/tag-transmission-test/httpserver-common-demo/target/httpserver-common-demo.jar > ${{ env.logDir }}/httpserver-common-demo.log 2>&1 &
+    - name: plugin change check
+      uses: ./.github/actions/common/plugin-change-check
```
不需要进行plugin-change-check，可以直接执行对应插件action

daizhenyu replied:
已修改

lilai23 commented on .github/actions/scenarios/tag-transmission/rocketmq/action.yml:
```diff
@@ -0,0 +1,82 @@
+      shell: bash
+      run: |
+        ps -ef | grep java
+        bash ./sermant-integration-tests/scripts/checkService.sh http://127.0.0.1:9056/rocketMqConsumer/queryRocketMqTag 120
```
其他启动的服务都加上健康检查校验

daizhenyu replied:
已修改

daizhenyu committed:
流量标签透传插件集成测试部分demo和集成测试脚本

流量标签透传插件集成测试部分demo和集成测试脚本-2

流量标签透传插件集成测试部分demo和集成测试脚本-2

流量标签透传插件集成测试部分demo和集成测试脚本-2
